src/App.js

```diff
@@ -12,6 +13,9 @@ function App() {
   const [results, setResults] = useState(null);
 
   useEffect(() => {
```

Comment:
The functionality you are trying to implement should be done in a different useEffect Hook which only runs when the component is first rendered. Your previous implementation was correct. The implementation that you have right now will lead to infinite re-rendering.

Reply:
I created a new useEffect Hook and add the blocks of code we implemented a few days ago.